src/main/java/seedu/todo/controllers/ListController.java

```diff
@@ -78,522 +80,147 @@ public void process(String input) throws ParseException {
-                    }
-                } else {
-                    return (List<Event>) null;
-                }
-            } else if (isCompleted) {
-                return db.getEventByRange((LocalDateTime) null, LocalDateTime.now());
-            } else {
-                return db.getEventByRange(LocalDateTime.now(), (LocalDateTime) null);
-            }
-        } else if (dateOn != null) { //by keyword found
-            return db.getEventByDate(dateOn);
-        } else {
-            return db.getEventByRange(dateFrom, dateTo);
-        }
-    }
-
-    /**
-     * Setting up Task view 
-     * 
-     * @param isCompleted
-     *            true if user request completed item
-     * @param listAllStatus
-     *            true if Status is not provided, isCompleted values will be ignored     
-     * @param dateOn
-     *            Date if user specify for a certain date
-     * @param dateFrom
-     *            Due date for Task or start date for Event
-     * @param dateTo
-     *            End date for Event  
-     * @param isDateProvided 
-     *            true if date is provided is any part of the input            
-     * @param isExactCommand
-     *            true if input is exactly the same as COMMAND_WORD
-     * @param listAll
-     *            true if CalendarItem Type not provided
-     * @param db
-     *            instance of TodoListDB db                              
-     */
-private List<Task> setupTaskView(boolean isCompleted, boolean listAllEventStatus, boolean listAllTaskStatus, LocalDateTime dateOn, LocalDateTime dateFrom,
-            LocalDateTime dateTo, boolean isDateProvided, boolean isExactCommand, boolean listAll, TodoListDB db) {
-        if (!listAllEventStatus) {
-            return null;
-        }
-        if (dateFrom == null && dateTo == null && dateOn == null) {
-            if (listAllTaskStatus) { // not specify
-                if (isExactCommand && isDateProvided == false) {
-                    if (listAll) {
-                        return db.getIncompleteTasksAndTaskFromTodayDate();
-                    } else {
-                        return db.getAllTasks();
-                    }
-                } else {
-                    return (List<Task>) null;
-                }
-            } else {
-                return db.getTaskByRangeWithStatus(dateFrom, dateTo, isCompleted, listAllTaskStatus);
-            }
-        } else if (dateOn != null) { //by keyword found
-            return db.getTaskByDateWithStatus(dateOn, isCompleted, listAllTaskStatus);
-        } else {
-            return db.getTaskByRangeWithStatus(dateFrom, dateTo, isCompleted, listAllTaskStatus);
-        }
-    }
-    
-    /** ================ PARSING METHODS ================== **/
-    
-    /**
-     * Parse a natural date into a LocalDateTime object.
-     * 
-     * @param natural
-     * @return LocalDateTime object
-     */
-    private LocalDateTime parseNatural(String natural) {
```

Comment:
@Candiie hmm i redid the controller as it is too messy. Agree that the changes is drastic but it looks more clean now? I suppose?